docs/FAQs.md

```diff
@@ -13,6 +13,122 @@ After you set up a repository, you can substitute in any tools you'd like.
+     So you'll be removing `tsup` in favour of [`ncc`](https://github.com/vercel/ncc) which compiles output into a single JS file.
+     Delete `tsup.config.ts` then execute the following commands:
+
+   ```bash
+   pnpm remove tsup
+   pnpm add @vercel/ncc -D
+   pnpm add @actions/core -P
+   ```
+
+   - We also added the [`@actions/core`](https://github.com/actions/toolkit/tree/master/packages/core) package, used for building GitHub Actions.
+     Now we need to update the `build` script in our `package.json`:
+
+   ```diff
+   -"build": "tsup",
+   +"build": "ncc build src/index.ts -o dist --license licenses.txt",
+   ```
+
+   - Our build now emits to the `dist` directory; so we'll want to avoid linting that directory by adding the following to `.eslintignore` and our `.prettierignore`:
+
+   ```diff
+   +dist
+   ```
+
+   - Rather than having to remember to compile each time, we'll update our precommit hook in `.husky/precommit` to build for us on each commit:
+
+   ```diff
+   +pnpm run build
+   +git add dist
+   npx lint-staged
+   ```
+
+   - on the off chance someone isn't running husky, we'll add `.github/workflows/check-dist.yml` ([based on this](https://github.com/actions/typescript-action/blob/main/.github/workflows/check-dist.yml)) to ensure `dist` is up to date:
+
+   ```yml
+   jobs:
+     check-dist:
+       name: check dist
+       runs-on: ubuntu-latest
+
+       steps:
+         - uses: actions/checkout@v4
+         - uses: ./.github/actions/prepare
+
+         - id: build
+           name: Build dist directory
+           run: pnpm run build
+
+         # This will fail the workflow if the PR wasn't created by Dependabot.
+         - id: diff
+           name: Compare directories
+           run: |
+             if [ "$(git diff --ignore-space-at-eol --text dist/index.js | wc -l)" -gt "0" ]; then
+               echo "Detected uncommitted changes after build."
+               echo "You may need to run 'pnpm run build' locally and commit the changes."
+               echo ""
+               echo "See diff below:"
+               echo ""
+               git diff --ignore-space-at-eol --text dist/index.js
+               echo ""
+               # say this again in case the diff is long
+               echo "You may need to run 'pnpm run build' locally and commit the changes."
+               echo ""
+               exit 1
+             fi
+
+         # If `dist/` was different than expected, and this was not a Dependabot
+         # PR, upload the expected version as a workflow artifact.
+         - id: upload
+           if: ${{ failure() && steps.diff.outcome == 'failure' }}
+           name: Upload artifact
+           uses: actions/upload-artifact@v4
+           with:
+             name: dist
+             path: dist/
+
+   name: Check transpiled index.js in dist is up to date
+
+   on:
+     pull_request: ~
+
+     push:
+       branches:
+         - main
+
+   permissions:
+     contents: read
+   ```
```

Comment:
This is probably the biggest <code>```suggestion</code> I've ever posted 😄 

```suggestion

    - On the off chance someone isn't running husky, we'll add a workflow to ensure `dist` is up to date:

      <details>
          <summary><code>.github/workflows/dist.yml</code></summary>

      ```yml
      jobs:
        check-dist:
          name: check dist
          runs-on: ubuntu-latest

          steps:
            - uses: actions/checkout@v4
            - uses: ./.github/actions/prepare

            - name: Build dist directory
              run: pnpm run build

            - name: Compare directories
              run: |
                if [ "$(git diff --ignore-space-at-eol --text dist/index.js | wc -l)" -gt "0" ]; then
                  echo "Detected uncommitted changes after build."
                  echo "You may need to run 'pnpm run build' locally and commit the changes."
                  echo ""
                  echo "See diff below:"
                  echo ""
                  git diff --ignore-space-at-eol --text dist/index.js
                  echo ""
                  # say this again in case the diff is long
                  echo "You may need to run 'pnpm run build' locally and commit the changes."
                  echo ""
                  exit 1
                fi

      name: Check built dist/ is up to date

      on:
        pull_request: ~

        push:
          branches:
            - main

      permissions:
        contents: read
      ```

       </details>
```